pkg/e2e/infra/infra.go

```diff
@@ -197,232 +223,63 @@ var _ = g.Describe("[Feature:Machines] Managed cluster should", func() {
+		scaleMachineSet(machineSet2.GetName(), 3)
+		e2e.WaitForMachineSet(client, machineSet.GetName())
+		e2e.WaitForMachineSet(client, machineSet2.GetName())
+
+		g.By("Deleting the second MachineSet")
+		err = client.Delete(context.Background(), machineSet2)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		e2e.WaitForMachineSetDelete(client, machineSet2)
 	})
 
 	g.It("drain node before removing machine resource", func() {
-		var err error
-		client, err := e2e.LoadClient()
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By("checking existing cluster size")
-		existingClusterSize, err := getClusterSize(client)
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		delObjects := make(map[string]runtime.Object)
+		g.By("Create a machine for node about to be drained")
 
-		defer func() {
-			// Remove resources
-			for _, obj := range delObjects {
-				if err := deleteObject(client, obj); err != nil {
-					glog.Errorf("[cleanup] error deleting object: %v", err)
-				}
-			}
-			// TODO(jchaloup): we need to make sure this gets called no matter what
-			// and waits until all labeled nodes are gone. Though, it it does not
-			// happend in the timeout set, it will not happen ever.
-			err := waitUntilNodesAreDeleted(client, []runtimeclient.ListOption{runtimeclient.MatchingLabels(nodeDrainLabels)})
-			o.Expect(err).NotTo(o.HaveOccurred())
-		}()
-
-		g.By("Taking the first worker machineset (assuming only worker machines are backed by machinesets)")
-		machinesets := mapiv1beta1.MachineSetList{}
-		err = wait.PollImmediate(e2e.RetryMedium, e2e.WaitShort, func() (bool, error) {
-			if err := client.List(context.TODO(), &machinesets); err != nil {
-				glog.Errorf("Error querying api for machineset object: %v, retrying...", err)
-				return false, nil
-			}
-			if len(machinesets.Items) < 1 {
-				glog.Errorf("Expected at least one machineset, have none")
-				return false, nil
-			}
-			return true, nil
-		})
-		o.Expect(err).NotTo(o.HaveOccurred())
-
-		g.By("Creating two new machines, one for node about to be drained, other for moving workload from drained node")
-		// Create two machines
-		randID := e2e.RandomString()
-		machine1 := machineFromMachineset(&machinesets.Items[0])
-		machine1.Name = "machine1-" + randID
-
-		var machine2 *mapiv1beta1.Machine
-		err = func() error {
-			if err := client.Create(context.TODO(), machine1); err != nil {
-				return fmt.Errorf("unable to create machine %q: %v", machine1.Name, err)
-			}
-			delObjects["machine1"] = machine1
-
-			machine2 = machineFromMachineset(&machinesets.Items[0])
-			machine2.Name = "machine2" + randID
+		// Add node draining labels to params
+		for k, v := range nodeDrainLabels {
+			machineSetParams.Labels[k] = v
+		}
 
-			if err := client.Create(context.TODO(), machine2); err != nil {
-				return fmt.Errorf("unable to create machine %q: %v", machine2.Name, err)
-			}
-			delObjects["machine2"] = machine2
+		machine, err := e2e.CreateMachine(client, machineSetParams)
```

Comment:
seems this is only creating one machine with `nodeDrainLabels`. So there's only one machine which satisfies the `replicationControllerWorkload` `NodeSelector`. So when draining the eviction API won't be able to find another suitable node.
Instead of creating a new machine, couldn't we use 2 machines of the ones created by the beforeEach machineSet and apply them the `nodeDrainLabels`? So
- Fetch 2 machines from machineSet
- Apply `nodeDrainLabels`
- Run workload (same as now)
- Delete one machine (same as now)
- verifyNodeDraining (same as now)